wordpress-mobile/WordPressUI-iOS
# Making FancyAlert scrollable.

This PR makes the FancyAlert view scrollable when its content is taller than the screen height. This change is particularly useful for devices with small screens, for landscape usage, and bigger font sizes (accessibility).

![fancy-scrolling](https://user-images.githubusercontent.com/9772967/51028958-3ed57e00-1595-11e9-8823-3405575e8bcf.gif)

I simplified the overall layout removing some unnecessary views and constraints, and added the scroll view inside the wrapper view, so the whole content is scrollable. At the same time, I made an effort to keep the previous layout, sizes and paddings, while making it look better with bigger text sizes.

I thought on leaving the buttons fixed at the bottom but it didn't play well in some scenarios, like many buttons with big text size on a small screen, or landscape mode.

![simulator screen shot - iphone 5s - 2019-01-11 at 10 23 57](https://user-images.githubusercontent.com/9772967/51028534-ebaefb80-1593-11e9-9531-677dddff52f1.png)

I tested all the instances of the FancyView used on WPiOS that I found, them being:
- **Notification Primer** (ask for notification permissions) 
- **Quick Start** (First time creating a site)
- **Gutenberg Alert** (Opening Gutenberg post with Aztec)
- **Async post** (Publishing a post while still uploading media)
- **Save for later** (First time using Save for later feature)
- **Site address helper** (Login by email screen helper)
- **Aztec verify email** (Publishing a post without email verified)

Also tested on iPad, iPhone 5S (simulator), iPhone XS.

**Known Issues:**
- The FancyAlert is not responding well to dynamically changing the font size. Some labels stay with the previous size. This is quite challenging to fix with the current configuration so I decided to leave it for the future.

- Button with long text will have the text truncated with bigger font sizes. I also prefer to leave this detail for a second revision.

**A question**
There is a `Title Accessory Button` that I didn't see being used on any of the FancyAlert instances. Is it worth keeping?

**To test:**
- Checkout `issue/fancy-alert-height` branch on the main [`WPiOS` project](https://github.com/wordpress-mobile/WordPress-iOS/pull/10798).
- Open some of the Fancy Alert instances and check that they show correctly.
  - I recommend to test `Gutenberg Alert` and `Site address helper` that are easy to show.
- Test on devices with different screen sizes (small iPhone, big iPhone, iPad)
- Test with Accessibility Text Sizes (iOS Settings -> General -> Accessibility -> Larger Text)

This component is probably also used on `WCiOS`. Might it be worth testing it there too?
cc @mindgraffiti
